pkg/vz/network_darwin.go

```diff
@@ -116,13 +109,12 @@ func (c *qemuPacketConn) Write(b []byte) (int, error) {
 		return 0, err
 	}
 
-	start := 0
-	for start < size {
-		nw, err := c.Conn.Write(b[start:])
+	for len(b) != 0 {
+		n, err := c.Conn.Write(b)
 		if err != nil {
 			return 0, err
 		}
-		start += nw
+		b = b[n:]
 	}
```

Comment:
The write change is not related to the fix - please split it to another PR. We don't want to mix refactoring with bug fixes.

Reply:
https://github.com/lima-vm/lima/pull/2923